flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/OperatorEventDispatcherImpl.java

```diff
@@ -66,6 +78,14 @@ void dispatchEventToHandlers(
             throw new FlinkException("Could not deserialize operator event", e);
         }
 
+        if (evt instanceof CloseGatewayEvent) {
+            OperatorEventGatewayImpl gateway = getOperatorEventGateway(operatorID);
+            gateway.sendEventToCoordinator(
+                    new AcknowledgeCloseGatewayEvent((CloseGatewayEvent) evt));
```

Comment:
Is it possible to have `sendEventToCoordinator()` be called concurrently with `dispatchEventToHandlers()`, such that operator sends an event after `AcknowledgeCloseGatewayEvent` is sent but before the gateway is closed? Can this be a problem?

Reply:
According to our offline discussion, I'll update the closing gateway process, to close gateways before sending `AcknowledgeCloseGatewayEvent`, so as to make sure there will be no event reaching the OC after the `AcknowledgeCloseGatewayEvent`.